sdk/src/Core/Amazon.Runtime/Credentials/Internal/IdentityResolvers/DefaultAWSTokenIdentityResolver.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using Smithy.Identity.Abstractions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Amazon.Runtime.Credentials.Internal.IdentityResolvers
+{
+    /// <summary>
+    /// A resolver that provides a bearer token identity.
+    /// </summary>
+    public class DefaultAWSTokenIdentityResolver : IIdentityResolver
+    {
+        private readonly IAWSTokenProvider _tokenProvider;
+
+        public DefaultAWSTokenIdentityResolver() : this(null) { }
+
+        public DefaultAWSTokenIdentityResolver(string profileName)
+            => _tokenProvider = new AWSTokenProviderChain(new ProfileTokenProvider(profileName));
+
+        /// <inheritdoc/>
+        public BaseIdentity ResolveIdentity()
+        {
+#if NETFRAMEWORK
+            if (_tokenProvider.TryResolveToken(out var token))
+            {
+                return token;
+            }
+
+            throw new AmazonClientException($"Failed to resolve bearer token in {nameof(DefaultAWSTokenIdentityResolver)}");
+#else
+            throw new System.NotSupportedException($"{nameof(ResolveIdentityAsync)} must be used in cross-platform .NET");
```

Comment:
Not liking this pattern of having methods exposed for cross platform .NET that then throw a runtime exception. We should either not have them or accept they have to do blocking async calls.

Reply:
Not a fan either. The SSO components only include the sync APIs when using the .NET Framework: https://github.com/aws/aws-sdk-net/blob/6cb3b06458449566cd85c4c79bb7849bcd4ceb48/sdk/src/Services/SSOOIDC/Custom/_bcl45%2Bnetstandard/AmazonSSOOIDCClient.Extension.cs#L26-L31

I'll update this to use a blocking call even though it's a bad practice.

Reply:
Added the blocking call in https://github.com/aws/aws-sdk-net/pull/3562/commits/8f86f7c0a8b212e51a0dfd29fe586b6fa8d34ae6#diff-96434be2538b1000e07b1470f39a00fcb6c772b0fb91ba06c74a65e277b24cce